### ISSUE TITLE
[Bug Report] Season dates

### ISSUE THREAD

**Issue by kubo6472** [PhlexPlexico/G5V#60]
Selecting a range of season dates, even though the latter date is in the future causes the season to not show up. There does not seem a way to un-select the range to turn it back into a one-day selection. The only resolution is to re-create the whole season.

**Comment by PhlexPlexico** [comment]
Technically you should be able to go back in time. The date picker is a little bit finnicky, and I was hoping that I could just use a single calendar but it seems that it is having a few too many issues. I can see there's also an error where it's giving invalid dates. Will look into this.

**Comment by PhlexPlexico** [comment]
Seems this was an API error actually. I've been playing around with some of the database connections. I think dates and everything should work and you shouldn't have to restart a season to get it to change dates. If you have a range selected, just click out of the calendar and click back in, and select your start date, then you should be able to select the end date. If you have a start date selected already, then you simply have to select an arbitrary end date, then you can select another start date. It seems the calendar remembers the last state it's in, so it could be confusing as to why it doesn't really select a "start date" after you try and de-select.

**Comment by kubo6472** [comment]
The problem still remains, as the season with any end date is basically invisible on my end. Tried everything, but the main problem is one non-admin acc on two devices. The solution, is to have a season with just a start date and no end date. But there's no way to "clear" the cal, to just have the start date specified and the season visible to not have to give super admin to every single acc that's supposed to see and use a season with an end date specified.

**Comment by PhlexPlexico** [comment]
Hm... I'm still having troubles trying to get what you mean and reproduce it on the front-end. Here's a video of what I'm doing and having it show up on a regular, non super admin account: https://youtu.be/VhmZD_Be_Qg

With the calendar in the seasons window, it will always alternate between start date -> end date -> start date... when you select in the calendar. So an alternative would be to just select a date until it just has the date, and not "2 selected", that will be your start date. I'm not sure where else you're not seeing the season pop up, since I can see it in the match create form as well... Sorry!

**Comment by PhlexPlexico** [comment]
However I did notice that the `/myseasons/available` was behaving rather oddly and not showing the proper seasons. I did fix that so you should only find your seasons on the match creation now, if the end date >= current date or end date is null.

**Comment by kubo6472** [comment]
Wow, thanks for the visual. I'll bump everything to the latest master and will let you know.

**Comment by PhlexPlexico** [comment]
Just a heads up, for G5API I'm currently testing out the `match_check_cleanup` branch, since it was a bit of code cleanup/a bit more transactional. Right now the change only exists in there because I want to wait and see how long it takes before the connection pool is poisoned, if at all. So checkout that branch and you should have the change available in G5API :)

Anytime!

**Comment by kubo6472** [comment]
Managed to get my problem [rec'd](https://wamis.ml/3xuIyIM). That I can't get it back to "ongoing" which probably isn't a big deal for me, since I can create matches for this season just fine. The problem is, my mate is using other super admin acc. but he can not access this season afaic.

**Comment by PhlexPlexico** [comment]
Ah yes, this seems to be a problem with the API. Since it strips any null values out of an update, it removes updating and setting the end date to null. I just pushed the change to match-check-cleanup, so pull from there and let me know if that fixes it.

As for the super admin issue - does he have the badge next to his profile name? It should show any active seasons for him when creating a match!

**Comment by kubo6472** [comment]
Yep, he has. Will let you know later about both of those on the latest update from that branch and test the admin season thing further.

**Comment by kubo6472** [comment]
The match clean checkup branch broke our BO2 vetoing. So I'm downgrading it back to master. See if that fixes our problem. The calendar problem seems to be fixed. Thanks.

**Comment by PhlexPlexico** [comment]
That's weird... There shouldn't be any changes from BO2 compared to what's in there since there is only improvements to the season calendar. Once I merge it in maybe it will fix it. What exactly "broke" in the BO2 vetoes?

**Comment by kubo6472** [comment]
![image](https://user-images.githubusercontent.com/26146502/116454121-f161a580-a85f-11eb-90c4-ffd781c1a7a4.png)
there is... we need

**Comment by PhlexPlexico** [comment]
Ah, that seems to be a get5 specific thing, nothing that I can exactly take care of on the web panel, sorry :(

**Comment by kubo6472** [comment]
I don't understand then: https://github.com/splewis/get5/wiki/Map-Vetoes#bo2-veto this should be the case then.

**Comment by kubo6472** [comment]
From what I get. get5 supports the veto we need in the BO2 format. But after creating that from the G5V, we get BO3 style veto.

**Comment by PhlexPlexico** [comment]
Okay so this is interesting. I made a change sometime ago with how the map setup is structured, and I think I may have backed out the change during a merge. In the match config there should be a `num_maps` call, but instead it's giving `maps_to_win`. The match cleanup branch in G5API should have the fix for it. Let me know if that works as intended or not.
